tests/integration_tests/modules/test_user_events.py

```diff
@@ -0,0 +1,104 @@
+"""Test user-overridable events.
+
+This is currently limited to applying network config on BOOT events.
+Since we reapply the same config on reboot, verification is limited to
+checking log messages.
+
+Because the supported/default events are datasource specific, the datasource
+is being limited to LXD containers/vms (NoCloud).
+For NoCloud, BOOT_NEW_INSTANCE and BOOT are supported, but only
+BOOT_NEW_INSTANCE is default.
```

Comment:
These tests look much better now, thank you! (But this docstring is now out-of-date. :runner::dash:)